### PR TITLE
Fix task resource unmarshal error case

### PR DESCRIPTION
**Pull request by fenxiong**
### Summary

Previously task resource unmarshal error was ignored rather than returned. This PR fixes it.

Also fixes unit tests that start failing on mac after exposing the error. I had to fix a lot of state files used in state manager unit tests because they were using cgroup, but cgroup is not supported on non-linux platform.

### Testing

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/taskresource/types/types.go [hunk 1]
```diff
@@ -51,35 +51,43 @@ func (rm *ResourcesMap) UnmarshalJSON(data []byte) error {
...
 		}
 	}
 	*rm = result
 	return nil
 }
 
+func unmarshalResource(key string, value json.RawMessage, result map[string][]taskresource.TaskResource) error {
+	switch key {
+	case CgroupKey:
+		if err := unmarshlCgroup(key, value, result); err != nil {
```
Comment (petderek) [inline agent/taskresource/types/types.go]: these could all be simplified to:

return umarshalThing(key,value,result)

```
// this...
if err != nil {
    return err
} else {
    return nil
}

// ...is the same as
return err

File: agent/taskresource/cgroup/cgroup_unsupported.go [hunk 2]
```diff
@@ -113,12 +113,12 @@ func (c *CgroupResource) GetTerminalReason() string {
 
 // MarshalJSON marshals CgroupResource object
 func (c *CgroupResource) MarshalJSON() ([]byte, error) {
-	return nil, errors.New("unsupported platform")
```
Comment (petderek) [inline agent/taskresource/cgroup/cgroup_unsupported.go]: We'd want this to fail loudly on windows if this path got crossed, right?

Comment (fenxiong) [reply agent/taskresource/cgroup/cgroup_unsupported.go]: yeah. keeping the error and fixes the state file instead

File: agent/taskresource/types/types.go (hunk 1 above)
Comment (fenxiong) [reply agent/taskresource/types/types.go]: updated

**Pull request merged**